castervoice/lib/ccr/core/text_manipulation.py

```diff
@@ -0,0 +1,254 @@
+                       Function(text_manipulation_functions.move_until_phrase,
+                       dict(right_character="phrase"),
+                       left_right="right", cursor_behavior="texstudio"))]),
+              rdescript="move to chosen character to the right of the cursor"),
+        "grab <lease_ross> [<number_of_lines_to_search>] <dictation>":
+            R(ContextAction(default=Function(text_manipulation_functions.select_phrase, 
+            dict(dictation="phrase", lease_ross="left_right"), 
+            cursor_behavior="standard"), actions=[(AppContext("texstudio"),
+            Function(text_manipulation_functions.select_phrase,
+                       dict(dictation="phrase", lease_ross="left_right"),
+                       cursor_behavior="texstudio"))]),
+                 rdescript="select chosen phrase"),
+        "grab lease [<number_of_lines_to_search>] <left_character>":
+            R(ContextAction(default=Function(text_manipulation_functions.select_phrase,
+            dict(left_character="phrase"), left_right="left",
+            cursor_behavior="standard"), actions=[(AppContext("texstudio"),
+            Function(text_manipulation_functions.select_phrase, dict(left_character="phrase"), left_right="left",
+            cursor_behavior="texstudio"))]),
+            rdescript="select chosen character to the left"),
+        "grab ross [<number_of_lines_to_search>] <right_character>":
+            R(ContextAction(default=Function(text_manipulation_functions.select_phrase, dict(right_character="phrase"), 
+            left_right="right", cursor_behavior="standard"), actions=[(AppContext("texstudio"),
+            Function(text_manipulation_functions.select_phrase, dict(right_character="phrase"), left_right="right",
+            cursor_behavior="texstudio"))]),
+            rdescript="select chosen character to the right"),
+        
+        "grab <lease_ross> [<number_of_lines_to_search>] until <dictation> ":
+            R(ContextAction(default=Function(text_manipulation_functions.select_until_phrase, 
+            dict(dictation="phrase", lease_ross="left_right"), 
+            cursor_behavior="standard"), actions=[(AppContext("texstudio"),
+            Function(text_manipulation_functions.select_until_phrase, dict(dictation="phrase", lease_ross="left_right"), 
+            cursor_behavior="texstudio"))]),
+                 rdescript="select until chosen phrase (inclusive)"),
+        "grab lease [<number_of_lines_to_search>] until <left_character>":
+            R(ContextAction(default=Function(text_manipulation_functions.select_until_phrase, dict(left_character="phrase"), 
+            left_right="left", cursor_behavior="standard"), actions=[(AppContext("texstudio"),
+            Function(text_manipulation_functions.select_until_phrase, dict(left_character="phrase"), 
+            left_right="left", cursor_behavior="texstudio"))]),
+            rdescript="select left until chosen character"),
```

Comment:
Add "Text Manipulation:" to the rdescripts and remove those for which the spec tells you everything you need to know about what it does.